.github/workflows/main.yml

```diff
@@ -45,7 +56,9 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-pip-
             ${{ runner.os }}-
+
       - name: Install dependencies
+        if: matrix.modules.module == 'main'
```

Comment:
The `module main` syntax is making the complex quite complex, how about just a add a new job step?

Reply:
I am not sure how to add a job step to separate the two tasks. Is there any more instructions?

Reply:
we have the `build` job: https://github.com/asyml/forte/blob/master/.github/workflows/main.yml#L12 and the `docs` job: https://github.com/asyml/forte/blob/master/.github/workflows/main.yml#L134

Also check here: https://github.com/asyml/forte-wrappers/blob/main/.github/workflows/main.yml, we have `build`, `pytest` and `docs`